src/tasks/minions/farmingActivity.ts

```diff
@@ -0,0 +1,426 @@
+import createReadableItemListFromBank from '../../lib/util/createReadableItemListFromTuple';
+import itemID from '../../lib/util/itemID';
+
+export default class extends Task {
+	async run({
+		plantsName,
+		patchType,
+		getPatchType,
+		quantity,
+		upgradeType,
+		userID,
+		channelID,
+		planting,
+		duration,
+		currentDate
+	}: FarmingActivityTaskOptions) {
+		const user = await this.client.users.fetch(userID);
+		const currentFarmingLevel = user.skillLevel(SkillsEnum.Farming);
+		const currentWoodcuttingLevel = user.skillLevel(SkillsEnum.Woodcutting);
+		let baseBonus = 1;
+		let bonusXP = 0;
+		let plantXp = 0;
+		let harvestXp = 0;
+		let compostXp = 0;
+		let checkHealthXp = 0;
+		let rakeXp = 0;
+		let woodcuttingXp = 0;
+		let payStr = '';
+		let wcStr = '';
+		let rakeStr = '';
+		let plantingStr = '';
+		let alivePlants = 0;
+		let chopped = false;
+		let farmingXpReceived = 0;
+		let chanceOfDeathReduction = 1;
+		let cropYield = 0;
+		let lives = 3;
+		let bonusXpMultiplier = 0;
+		let farmersPiecesCheck = 0;
+
+		const plant = Farming.Plants.find(plant => plant.name === plantsName);
+		const userBank = user.settings.get(UserSettings.Bank);
+
+		if (bankHasItem(userBank, itemID('Magic secateurs'))) {
+			baseBonus += 0.1;
+		}
+
+		if (
+			bankHasItem(userBank, itemID('Farming cape')) ||
+			bankHasItem(userBank, itemID('Farming cape(t)'))
+		) {
+			baseBonus += 0.05;
+		}
+
+		if (upgradeType === 'compost') compostXp = 18;
+		if (upgradeType === 'supercompost') compostXp = 26;
+		if (upgradeType === 'ultracompost') compostXp = 36;
+
+		// initial lives = 3. Compost, super, ultra, increases lives by 1 respectively and reduces chanceofdeath as well.
+		// Payment = 0% chance of death
+		if (patchType.lastUpgradeType === 'compost') {
+			lives += 1;
+			chanceOfDeathReduction = 1 / 2;
+		} else if (patchType.lastUpgradeType === 'supercompost') {
+			lives += 2;
+			chanceOfDeathReduction = 1 / 5;
+		} else if (patchType.lastUpgradeType === 'ultracompost') {
+			lives += 3;
+			chanceOfDeathReduction = 1 / 10;
+		}
+
+		if (patchType.lastPayment) chanceOfDeathReduction = 0;
+
+		// check bank for farmer's items
+		if (bankHasItem(userBank, itemID(`Farmer's strawhat`))) {
+			bonusXpMultiplier += 0.004;
+			farmersPiecesCheck++;
+		}
+		if (
+			bankHasItem(userBank, itemID(`Farmer's jacket`)) ||
+			bankHasItem(userBank, itemID(`Farmer's shirt`))
+		) {
+			bonusXpMultiplier += 0.008;
+			farmersPiecesCheck++;
+		}
+		if (bankHasItem(userBank, itemID(`Farmer's boro trousers`))) {
+			bonusXpMultiplier += 0.006;
+			farmersPiecesCheck++;
+		}
+		if (bankHasItem(userBank, itemID(`Farmer's boots`))) {
+			bonusXpMultiplier += 0.002;
+			farmersPiecesCheck++;
+		}
+		if (farmersPiecesCheck === 4) bonusXpMultiplier += 0.005;
+
+		let loot = {
+			[itemID('Weeds')]: 0
+		};
+
+		delete loot[itemID('Weeds')];
+
+		if (!patchType.patchPlanted) {
+			if (!plant) return;
```

Comment:
whats this random `return`? if this is a scenario where you never ever expect this to happen, use something like this: `this.client.wtf('Had no plant available in farmingActivity.')`

Reply:
same applies everywhere.